internal/clients/etherscan/pubkey-finder_test.go

```diff
@@ -0,0 +1,109 @@
+package etherscan
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestGetFromResultHash(t *testing.T) {
+	assert := assert.New(t)
+
+	mockTxResult := []txResult{
+		{From: "address1", Hash: "aaa111"},
+		{From: "address2", Hash: "bbb222"},
+		{From: "address3", Hash: "ccc333"},
+	}
+
+	type args struct {
+		address string
+		txList  *txList
+	}
+
+	testCases := []struct {
+		name    string
+		args    args
+		want    common.Hash
+		wantErr bool
+	}{
+		{
+			"empty-transaction-list",
+			args{
+				"should-not-matter",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  []txResult{},
+				},
+			},
+			common.Hash{},
+			true,
+		},
+		{
+			"match-transaction-1",
+			args{
+				"address1",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  mockTxResult,
+				},
+			},
+			common.HexToHash("aaa111"),
+			false,
+		},
+		{
+			"match-transaction-2",
+			args{
+				"address2",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  mockTxResult,
+				},
+			},
+			common.HexToHash("bbb222"),
+			false,
+		},
+		{
```

Comment:
I don't this this test case is needed checking that it returns the correct value is covered in both tests cases above. E.g. first and not first.